fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2174,7 +2181,17 @@ public Range getUnbuiltRange() {
                 ImmutableMap.Builder<String, IndexState> indexStateMapBuilder = ImmutableMap.builder();
 
                 for (KeyValue kv : list) {
-                    String indexName = isSubspace.unpack(kv.getKey()).getString(0);
+                    String indexName;
+                    if (indexStateIsStoredInOldFormat(storeHeader)) {
+                        indexName = isSubspace.unpack(kv.getKey()).getString(0);
+                    } else {
+                        Index index = getRecordMetaData().getIndex(isSubspace.unpack(kv.getKey()).get(0));
```

Comment:
Isn't `getIndex` by name rather than by subspace key? And I think it throws an error on absence instead of returning `null`.

Reply:
Right, the pre-existing version is by name. I've introduced a new one which returns `null`.

Reply:
Okay, now I see.

I think we should probably at least call the new one `getIndexByKey` or `findIndexBySubspaceTupleKey` or something--something to differentiate it from `getIndex(String)` (especially as `String` is an `Object`).

It is...somewhat weird to me that these two versions of the function would have different semantics if the map is entry, but maybe it's sufficient if they are sufficiently documented. It still makes me somewhat uneasy.

Reply:
Sure, let me rename it to make it distinct.